### PR TITLE
Add vagrant linux build environment

### PR DESCRIPTION
**Pull request by destijl**
- Fix debian configs to work with new debhelper
- Make postinstall work on old ubuntu
- Add vagrant configs for building on lucid and trusty
- Add script to install all build dependencies from source

This is a clean commit history version of https://github.com/google/grr/pull/105 see there for comment history.

### CODE REVIEW

File: vagrant/install_from_src.sh [hunk 1]
```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
```
Comment (grrrrrrrrr) [inline vagrant/install_from_src.sh]: I'd recommend a set +e here, there are so many things that could go wrong...

File: scripts/install_script_ubuntu.sh [hunk 2]
```diff
@@ -42,6 +49,14 @@ then
   GRR_TESTING=1;
 fi
 
+if [ "$1" == "--build-deps" ] || [ "$2" == "--build-deps" ]; then
```
Comment (grrrrrrrrr) [inline scripts/install_script_ubuntu.sh]: should we do this in a while / shift loop? install.sh --build-deps --localtest would already not work...

File: vagrant/install_from_src.sh (hunk 1 above)
Comment (destijl) [inline vagrant/install_from_src.sh]: Done. Exposed a couple of non-obvious bugs too.  Openssl changed their symlinking/hashing scheme!

File: scripts/install_script_ubuntu.sh (hunk 2 above)
Comment (destijl) [inline scripts/install_script_ubuntu.sh]: Added proper option parsing.

**Comment by destijl** [comment]
PTAL

**Comment by grrrrrrrrr** [comment]
We need to also update the documentation that the install script parameters have changed. Otherwise, LGTM

**Comment by destijl** [comment]
@grrrrrrrrr I'm going to merge this now, can address any comments on those last changes when I move it into the internal repo.

**Pull request merged**